contracts/src/external/HyperdriveTarget0.sol

```diff
@@ -35,12 +36,31 @@ abstract contract HyperdriveTarget0 is
 
     /// @notice Instantiates target0.
     /// @param _config The configuration of the Hyperdrive pool.
+    /// @param __adminController The admin controller that will specify the
+    ///        admin parameters for this contract.
     constructor(
-        IHyperdrive.PoolConfig memory _config
-    ) HyperdriveStorage(_config) {}
+        IHyperdrive.PoolConfig memory _config,
+        IHyperdriveAdminController __adminController
+    ) HyperdriveStorage(_config, __adminController) {}
 
     /// Admin ///
 
+    // TODO: This function doesn't do anything anymore and is only here for
+    // backwards compatability. This can be removed when the factory is upgraded.
+    //
+    /// @notice A stub for the old setPauser functions that doesn't do anything
+    ///         anymore.
+    /// @dev Don't call this. It doesn't do anything.
+    function setGovernance(address) external {}
```

Comment:
I'm not totally following this. Maybe we can do a walkthrough. 

One question, why not just have it revert if it is called?

Reply:
It’s called during deployAndInitialize, so that would break backward compatibility and require a new factory to be deployed